src/content/peerconnection/bandwidth/js/main.js

```diff
@@ -288,3 +314,43 @@ window.setInterval(() => {
     lastResult = res;
   });
 }, 1000);
+
+function syntheticVideoStream({width = 640, height = 480, signal} = {}) {
+  const canvas = Object.assign(
+      document.createElement('canvas'), {width, height}
+  );
+  const ctx = canvas.getContext('2d');
+  const stream = canvas.captureStream();
+
+  let count = 0;
+  setInterval(() => {
+    // Use relatively-prime multipliers to get a color roll
+    const r = Math.abs((count*2)%512 - 256);
+    const g = Math.abs((count*3)%512 - 256);
+    const b = Math.abs((count*5)%512 - 256);
+    ctx.fillStyle = `rgb(${r}, ${g}, ${b})`;
+    count += 1;
+    const boxSize=80;
+    ctx.fillRect(0, 0, width, height);
+    // Add some bouncing boxes in contrast color to add a little more noise.
+    const rContrast = (r + 128)%255;
+    const gContrast = (g + 128)%255;
+    const bContrast = (b + 128)%255;
```

Comment:
Did you perhaps mean 256 rather than 255 here?

Reply:
I woke up in the night and said to myself "%255 will only go to 254!" :-)
This was code I copied from RTCPeerConnection-helper.js in WPT; it was improper from before.

(Once this is landed, I intend to take the signal generator back into RTCPeerConnection-helper)